linux_os/guide/system/bootloader-grub2/uefi/grub2_uefi_kernel_trust_cpu_rng/oval/shared.xml

```diff
@@ -0,0 +1,65 @@
+<def-group>
+  <definition class="compliance" id="grub2_uefi_kernel_trust_cpu_rng" version="1">
+    {{{ oval_metadata("Ensure the kernel is configured to trust the CPU hardware random number generator.") }}}
+    <criteria operator="OR" >
+      <criteria operator="AND">
+        <criterion comment="check if the option is compiled in" test_ref="test_trust_cpu_rng_compiled_in" />
+        <criterion comment="check if the option is not overridden through a boot parameter" test_ref="test_trust_cpu_rng_boot_param_off" negate="true" />
+      </criteria>
+      <criterion comment="check if the option is configured as a boot parameter" test_ref="test_trust_cpu_rng_boot_param_on" />
+    </criteria>
+  </definition>
+
+  <local_variable id="var_kernel_config_file" comment="the config file for currently running kernel" datatype="string" version="1">
```

Comment:
Unfortunately, we can't detect identical OVAL objects and deduplicate them. The variable existing twice in the OVAL (although it is a "local" variable) creates problems, so you can work around this by duplicating it under a different name:

```suggestion
  <local_variable id="var_{{{ rule_id }}}_kernel_config_file" comment="the config file for currently running kernel" datatype="string" version="1">
```